Client/Frontend/Widgets/AutocompleteTextField.swift

```diff
@@ -163,14 +164,14 @@ class AutocompleteTextField: UITextField, UITextFieldDelegate {
     // Since the text has changed, remove the completion here, and textDidChange will fire the callback to
     // get the new autocompletion.
     func textField(_ textField: UITextField, shouldChangeCharactersIn range: NSRange, replacementString string: String) -> Bool {
-        lastReplacement = string
+        wasDeleteTyped = string.isEmpty
         return true
     }
 
     func setAutocompleteSuggestion(_ suggestion: String?) {
         let text = self.text ?? ""
 
-        guard let suggestion = suggestion, isEditing && markedTextRange == nil else {
+        guard let suggestion = suggestion, !wasDeleteTyped && isEditing && markedTextRange == nil else {
```

Comment:
setAutocompleteSuggestion is called by the searchDelegate when a new search term arrives. As long as we dont call `notifyTextChanged` unnecessarily I dont think we should need this.